src/main/java/com/mael/payroll/transactions/AddEmployeeTransaction.java

```diff
@@ -0,0 +1,8 @@
+package com.mael.payroll.transactions;
+
+import com.mael.payroll.PayrollDBFacade;
+
+public interface AddEmployeeTransaction {
+    void execute();
+    void addDB(PayrollDBFacade payrollDB);
```

Comment:
This adds temporal coupling.
If an instance of `PayrollDBFacade` is needed in order to have an `AddEmployeeTransaction` object work properly, I shouldn't have the need to have a two-step setup (i.e. create a new object _and_ call another setup method).

So I'd either pass it to the constructor of every implementing class of `AddEmployeeTransaction` or to `execute()` depending on which option gives me less of a bad feeling.